fix(linter): Fix incorrect `allowNames` config option name in `typescript/no-this-alias` rule.

connorshea opened the pull request:
This change corrects the configuration option name from `allowNames` to `allowedNames` in the `no-this-alias` rule of the linter. The previous name was seemingly a mistake, the original typescript-eslint rule uses allowedNames.

I've retained allowNames as an alias for backwards compatibility with existing configurations that set up their config based on the config docs we previously had.

See https://typescript-eslint.io/rules/no-this-alias/

Generated docs after this PR:

```md
## Configuration

This rule accepts a configuration object with the following properties:

### allowDestructuring

type: `boolean`

default: `true`

Whether to allow destructuring of `this` to local variables.


### allowedNames

type: `string[]`

default: `[]`

An array of variable names that are allowed to alias `this`.
```

camc314 commented:
### Merge activity

* **Dec 14, 6:50 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 14, 6:54 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
fix(linter): Fix incorrect `allowNames` config option name in `typescript/no-this-alias` rule. (#16852)

This change corrects the configuration option name from `allowNames` to `allowedNames` in the `no-this-alias` rule of the linter. The previous name was seemingly a mistake, the original typescript-eslint rule uses allowedNames.

I've retained allowNames as an alias for backwards compatibility with existing configurations that set up their config based on the config docs we previously had.

See https://typescript-eslint.io/rules/no-this-alias/

Generated docs after this PR:

```md
## Configuration

This rule accepts a configuration object with the following properties:

### allowDestructuring

type: `boolean`

default: `true`

Whether to allow destructuring of `this` to local variables.

### allowedNames

type: `string[]`

default: `[]`

An array of variable names that are allowed to alias `this`.
```
